projects/plugins/jetpack/modules/masterbar/profile-edit/profile-edit.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * WP-Admin Profile edit.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Hides profile fields for WordPress.com connected users.
+ *
+ * @param WP_User $user The current WP_User object.
+ */
+function jetpack_masterbar_hide_profile_fields( $user ) {
+	$connection_manager = new Connection_Manager( 'jetpack' );
+	if ( ! $connection_manager->is_user_connected( $user->ID ) ) {
+		// If this is a local user, show the default UX.
+		return;
+	}
+	$wp_kses_rule = array(
+		'a' => array(
+			'href'   => array(),
+			'rel'    => array(),
+			'target' => array(),
+		),
+	);
+	// Since there is no hook for altering profile fields, we will use CSS and JS.
+	$name_info_wpcom_link_message = sprintf(
+		/* translators: 1 link */
+		__( 'WordPress.com users can change their profile\\\'s basic details ( i.e., First Name, Last Name, Display Name, About ) in <a href="%1$s" target="_blank" rel="noopener noreferrer">WordPress.com Profile settings.</a>', 'jetpack' ),
+		'https://wordpress.com/me'
+	);
+	$contact_info_wpcom_link_message = sprintf(
+		/* translators: 1 link */
+		__( 'WordPress.com users can change their profile\\\'s email & website address in <a href="%1$s" target="_blank" rel="noopener noreferrer">WordPress.com Account settings.</a>', 'jetpack' ),
+		'https://wordpress.com/me/account'
+	);
+	?>
+	<script>
+		document.addEventListener( 'DOMContentLoaded', function() {
+			// Field to be hidden.
+			var fieldsToHide = '#your-profile .user-first-name-wrap, #your-profile .user-last-name-wrap, #your-profile .user-nickname-wrap, #your-profile .user-display-name-wrap, #your-profile .user-email-wrap, #your-profile .user-url-wrap, #your-profile .user-description-wrap';
+			document.querySelectorAll(fieldsToHide).forEach( element => element.classList.add( 'hidden') );
+
+			// Name Info.
+			var nameInfo                    = document.querySelector( '.user-first-name-wrap' ).closest( 'table' );
```

Comment:
Question: Why the `#your-profile` selector is not used here? If not needed, can it be removed from `fieldsToHide`?

Reply:
> If not needed, can it be removed from fieldsToHide?

It is redundant at the moment, done! 2bc3298